Parser/Types/TypeAnalyze.cs

```diff
@@ -26,14 +28,24 @@ public override NType Visit (NProgram p)
       Visit (d.Vars); return Visit (d.Funcs);
    }
 
+   public override NType Visit (NConstDecl c) {
+      mSymbols.Consts.Add (c);
+      return Void;
+   }
+
    public override NType Visit (NVarDecl d) {
       mSymbols.Vars.Add (d);
       return d.Type;
    }
 
    public override NType Visit (NFnDecl f) {
       mSymbols.Funcs.Add (f);
-      return f.Return;
+      if (mSymbols.Find (f.Name.Text) is not NVarDecl) {

```

Comment:
What is this if check for (not NVarDecl). Isn't it already known that f is a function?